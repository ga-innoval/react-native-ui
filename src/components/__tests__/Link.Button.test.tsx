import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react-native';
import { LinkButton } from '../Link.Button';

const eventData = {
  nativeEvent: {
    pageX: 20,
    pageY: 30,
  },
};

describe('<LinkButton />', () => {
  let component: any = null;
  const mockFn = jest.fn();

  beforeEach(() => {
    component = render(<LinkButton text="Press me" onPress={mockFn} />);
  });

  it('should render correctly', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call onPress function', () => {
    const button = screen.getByText(/press/i);
    fireEvent.press(button, eventData);
    expect(mockFn).toBeCalledTimes(1);
  });
});
