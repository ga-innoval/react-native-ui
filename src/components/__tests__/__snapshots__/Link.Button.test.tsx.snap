// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinkButton /> should render correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "flexDirection": "row",
        "marginBottom": 10,
      },
      null,
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#7A7878",
          "fontSize": 16,
        },
        {
          "color": "#748EEC",
        },
      ]
    }
  >
    Press me
  </Text>
  <Text />
</View>
`;
